packages/sap-adapter/src/adapter.ts

```diff
@@ -0,0 +1,154 @@
+/*
+*                      Copyright 2023 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import _ from 'lodash'
+import {
+  FetchResult, AdapterOperations, DeployResult, InstanceElement, TypeMap, isObjectType,
+  DeployModifiers, FetchOptions,
+} from '@salto-io/adapter-api'
+import { client as clientUtils, config as configUtils, elements as elementUtils } from '@salto-io/adapter-components'
+import { logDuration } from '@salto-io/adapter-utils'
+import { logger } from '@salto-io/logging'
+import SAPClient from './client/client'
+import { SAPConfig, API_DEFINITIONS_CONFIG, FETCH_CONFIG, SAPApiConfig } from './config'
+import { FilterCreator, Filter, filtersRunner } from './filter'
+import commonFilters from './filters/common'
+import fieldReferencesFilter from './filters/field_references'
+import changeValidator from './change_validator'
+import { SAP } from './constants'
+import { paginate } from './client/pagination'
+
+const { createPaginator } = clientUtils
+const { generateTypes, getAllInstances } = elementUtils.swagger
+const log = logger(module)
+
+const { hideTypes, ...otherCommonFilters } = commonFilters
+
+export const DEFAULT_FILTERS = [
+  hideTypes,
+  // fieldReferencesFilter should run after all elements were created
+  fieldReferencesFilter,
+  ...Object.values(otherCommonFilters),
+]
```

Comment:
I think I copied this from a more complex example, this can be
```suggestion
export const DEFAULT_FILTERS = [
  // fieldReferencesFilter should run after all elements were created
  fieldReferencesFilter,
  ...Object.values(commonFilters),
]
```